backend/api/src/supabase-search-groups.ts

```diff
@@ -13,13 +13,20 @@ const bodySchema = z.object({
   limit: z.number().gt(0),
   fuzzy: z.boolean().optional(),
   yourGroups: z.boolean().optional(),
+  addingToContract: z.boolean().optional(),
+  newContract: z.boolean().optional(),
```

Comment:
In general, I think it's better to for the api to describe what it does, rather than current case you are using it for. Makes it easier to reuse, but also makes it easier to understand what is actually happening.

For example, instead of `addingToContract` it could be `hasModeratorOrAboveRole` and `newContract` could be `isNonPrivate`, etc.

Or maybe you could set filters to a list of visibilities: `['public', 'curated']`

Reply:
Yeah unfortunately hasModeratorOrAboveRole means something different if you're searching for groups to add to your contract vs just searching for groups in general.